imports/api/blocks/server/methods.js

```diff
@@ -434,20 +434,29 @@ Meteor.methods({
                     let chainStatus = Chain.findOne({chainId:block.block.header.chain_id});
                     let lastSyncedTime = chainStatus?chainStatus.lastSyncedTime:0;
                     let timeDiff;
-                    let blockTime = Meteor.settings.params.defaultBlockTime;
+                    let blockTime = 0;
+                    let res;
                     if (lastSyncedTime){
+                        let blockDiff = Meteor.settings.public.averageBlockTimeWindow;
+                        let pastBlockHeight = (height - blockDiff > 0) ? (height - blockDiff) : (height - 1);
+                        let getPastBlock = `${API}/blocks/${pastBlockHeight}`;
+                        try{
+                            res = HTTP.get(getPastBlock);
+                        }
+                        catch(e){
+                            console.log(e)
+                        }
+                        let pastBlockTime = new Date(JSON.parse(res?.content)?.block?.header?.time);
                         let dateLatest = new Date(blockData.time);
                         let dateLast = new Date(lastSyncedTime);
-                        let genesisTime = new Date(Meteor.settings.public.genesisTime);
                         timeDiff = Math.abs(dateLatest.getTime() - dateLast.getTime());
-                        // blockTime = (chainStatus.blockTime * (blockData.height - 1) + timeDiff) / blockData.height;
-                        blockTime = (dateLatest.getTime() - genesisTime.getTime()) / blockData.height;
+                        blockTime = (height - blockDiff > 0) ? ((dateLatest.getTime() - pastBlockTime.getTime()) / blockDiff) : (dateLatest.getTime() - pastBlockTime.getTime());
```

Comment:
Since the genesis time value had been removed from the settings.json file, we cannot calculate the average block time by using ```averageBlockTime = lastBlockTime - genesisTime / blockHeight```. To handle the calculations, I initially added averageBlockTimeWindow in settings.json file to calculate the average block time of last n blocks...

Reply:
However, I changed the approach and I've just updated the code to calculate average block time starting from the startHeight height set in params in settings.json file

Reply:
Ah... I have missed the `genesisTime` in the `default_settings.json`. It should be there as this is how the average block time is being calculated. Querying from chain is not feasible as the genesis could be huge. The size on `cosmoshub-4` can't even give a full response if we query from the api.

Reply:
I see! Alright I will update the code to include ```genesisTime``` calculations again.

Reply:
Updated, how does it do now?